src/jit/lower.cpp

```diff
@@ -3123,6 +3133,195 @@ void Lowering::LowerAdd(GenTreePtr* pTree, Compiler::fgWalkData* data)
 #endif // !_TARGET_ARMARCH_
 }
 
+//------------------------------------------------------------------------
+// LowerUDivUMod: transform GT_UDIV/GT_UMOD nodes with a const power of 2
+// divisor into GT_RSZ/GT_AND nodes.
+//
+// Arguments:
+//    tree:   pointer to the GT_UDIV/GT_UMOD node to be lowered
+
+void Lowering::LowerUDivUMod(GenTree* tree)
+{
+    GenTreeOp* divmod = tree->AsOp();
+    GenTree* divisor = divmod->gtOp2;
+
+    if (divisor->IsCnsIntOrI())
+    {
+        size_t divisorValue = static_cast<size_t>(divisor->gtIntCon.IconValue());
+
+        if (isPow2(divisorValue))
+        {
+            genTreeOps newOper;
+
+            if (divmod->gtOper == GT_UDIV)
```

Comment:
@sivarv Presumably the same is true about other data members which have get accessors - `gtType` and `gtOp1`?